### PR TITLE
MethylVI batch effect tutorial

### PR DESCRIPTION
**Pull request by ethanweinberger**
Tutorial accompanies https://github.com/scverse/scvi-tools/pull/2834

## Formatting

- [x] My tutorial has only one top-level (`#`) header

## Reproducibility

- [x] My tutorial works on Google Colab
- [x] My tutorial sets `scvi.settings.seed = 0` at the beginning of the notebook
- [x] My tutorial has been run and includes outputs (e.g. plots, tables)

## Other

- [x] Counts and normalized data should co-exist in the datasets, see the [API overview](https://docs.scvi-tools.org/en/stable/tutorials/notebooks/api_overview.html) for an example
- [x] For scRNA-seq data, normalization should be counts per median library size and then log1p transformed -- if not, a reason should be given (Note: not applicable to this tutorial as we're not using scRNA-seq)

### CODE REVIEW

File: scbs/MethylVI_batch.ipynb [hunk 1]
```diff
@@ -0,0 +1,468 @@
+{
```
Comment (canergen) [inline scbs/MethylVI_batch.ipynb:1]: <p>Make consistent with other tutorials.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scverse/scvi-tutorials/pull/363/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (canergen) [inline scbs/MethylVI_batch.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.        os.system(f"wget -q -O {mdata_path} https://figshare.com/ndownloader/files/49632108")
</pre><p>@ori can we use our download function instead?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scverse/scvi-tutorials/pull/363/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (canergen) [inline scbs/MethylVI_batch.ipynb:1]: <p>jointly = one model but treat both seperately not add both and get one output? Maybe write "we'll jointly model both".</p><p>How would <code>modalities={"categorical_covariate_keys": "mCG"}</code>look like for two different fields. Please put in the comment.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scverse/scvi-tutorials/pull/363/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (canergen) [inline scbs/MethylVI_batch.ipynb:1]: <p>Maybe show get_normalized_methylation or DE additionally to go beyond latent space analysis? See our other tutorials for examples.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scverse/scvi-tutorials/pull/363/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

**Comment by canergen** [comment]
See comments overall looks fine. @ori-kron-wis can you help make the style consistent with other tutorials (I assume removing notebook style commands). These might change rendering on the webpage, which can end up ugly.

File: scbs/MethylVI_batch.ipynb (hunk 1 above)
Comment (ethanweinberger) [reply scbs/MethylVI_batch.ipynb:1]: <p>Once this notebook is merged in I plan to create another tutorial for feature-level analysis</p><p> </p>

Comment (ethanweinberger) [reply scbs/MethylVI_batch.ipynb:1]: <p>@canergen what exactly should I change here? I took this block of code from the [Atlas integration tutorial[(<a href="https://docs.scvi-tools.org/en/stable/tutorials/notebooks/scrna/harmonization.html" target="_blank">https://docs.scvi-tools.org/en/stable/tutorials/notebooks/scrna/harmonization.html</a>) </p><p> </p>

Comment (canergen) [reply scbs/MethylVI_batch.ipynb:1]: <p>Thanks and my bad. It was more a note to Ori to double-check once. Resolved.</p>

Comment (canergen) [reply scbs/MethylVI_batch.ipynb:1]: <p>Can we have it in one notebook? I don't like bloating the number of tutorials. It's currently already quite painful to rerun and manually merge notebooks. Do you need different datasets for it?</p>

Comment (canergen) [reply scbs/MethylVI_batch.ipynb:1]: Maybe you post here the code if both have different categories and then I can advise what's the minimum information required.

Comment (ethanweinberger) [reply scbs/MethylVI_batch.ipynb:1]: For feature-level analysis I would prefer to use a different dataset (I don't have as good ground truths for the batch effect dataset). I'm happy to put both analyses in a single notebook though if that makes things easier for you?

Comment (ethanweinberger) [reply scbs/MethylVI_batch.ipynb:1]: @canergen what's the exact scenario you're thinking of?

I think I misunderstood your original comment, and I'm not sure what you're asking (was `if mCH and mCH` above a typo?). For example, by two different fields did you mean:

1. I have two categorical covariate fields, one if which lives in `mdata['mCG'].obs` and the other of which lives in `mdata['mCH'].obs`? In this case I believe the `MuDataCategoricalJointObsField` used in the `setup_mudata` function doesn't support this use case, and we would ask users to consolidate their categorical variables into one modality's `.obs` field. 
2. Alternatively, do you mean one argument to the `setup_mudata` function is stored in one modality (e.g. `batch_key` is in `mCG`) while the other is in a different field (e.g. `categorical_covariate_keys` is in `mCH`)? This case should be straightforward, and I can modify the notebook with corresponding instructions.

Comment (ori-kron-wis) [reply scbs/MethylVI_batch.ipynb:1]: not sure, instead of what?

**Comment by ori-kron-wis** [comment]
@canergen not so sure what is the problem/difference in style? care to give a few examples?

**Comment by ethanweinberger** [comment]
Hi @canergen. Just wanted to check in to see if you had updates re: my and @ori-kron-wis's previous comments.

File: scbs/MethylVI_batch.ipynb (hunk 1 above)
Comment (canergen) [reply scbs/MethylVI_batch.ipynb:1]: No if it's using different datasets, make it different notebooks.

**Review by canergen — APPROVED** [review]
Please replace wget with pooch.

File: scbs/MethylVI_batch.ipynb (hunk 1 above)
Comment (ori-kron-wis) [reply scbs/MethylVI_batch.ipynb:1]: replaces wget with pooch

**Pull request merged**